task/pylint/0.1/pylint.yaml

```diff
@@ -37,9 +37,16 @@ spec:
   steps:
     - name: lint
       image: docker.io/python:$(inputs.params.PYTHON)
-      workingDir: /workspace/source
+      workingDir: $(workspaces.source.path)
       script: |
         export PATH=$PATH:$HOME/.local/bin
-        pip install -r $(inputs.params.SOURCE_PATH)/$(inputs.params.REQUIREMENTS_FILE)
-        pip show pylint || echo "###\nWarning: Pylint is missing in your requirements\n###" && pip install pylint
+        if [ -n "$(inputs.params.REQUIREMENTS_FILE)" ] && [ -e "$(inputs.params.REQUIREMENTS_FILE)" ];then
```

Comment:
can we set the default value to `""` in `REQUIREMENTS_FILE` params at line no [36](https://github.com/tektoncd/catalog/pull/459/files#diff-4b6a867b7d4f0c4e83dbaa5d9ad0ba25L36) and allow user to pass the `requirements.txt` filepath in the `TaskRun` so that it is more user-controlled :thinking:

Reply:
well you can pass the REQUIREMENTS_FILE already, is what you want to not set it as default ?

Reply:
Yes

Reply:
it's both way user-controlled, isnt it ?

Reply:
yes it is now also user-controlled, just presented my thought on it :sweat_smile: